File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantInfo.java
package com.github.zxh.classpy.classfile.constant;

import com.github.zxh.classpy.classfile.ClassFileComponent;

/*
cp_info {
    u1 tag;
    u1 info[];
}
 */
public abstract class ConstantInfo extends ClassFileComponent {

    {
        u1("tag");
    }

    protected abstract String loadDesc(ConstantPool cp);
    
}

File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantInfo.kt
package com.github.zxh.classpy.classfile.constant;

import com.github.zxh.classpy.classfile.ClassComponent;

/*
cp_info {
    u1 tag;
    u1 info[];
}
 */
abstract class ConstantInfo : ClassComponent() {

    init {
        u1("tag");
    }

    abstract fun loadDesc(cp: ConstantPool): String;
    
}
